Cython/Build/Cythonize.py

```diff
@@ -139,6 +139,9 @@ def parse_args(args):
     parser.add_option('-X', '--directive', metavar='NAME=VALUE,...', dest='directives',
                       type=str, action='callback', callback=parse_directives, default={},
                       help='set a compiler directive')
+    parser.add_option('-E', '--compile-time-env', metavar='NAME=VALUE,...', dest='compile_time_env',
+                      type=str, action='callback', callback=Options.parse_compile_time_env, default={},
```

Comment:
I can see in the surrounding code where this is coming from, but it seems wrong to say `type=str` when the actual value is a dict. And I'm also not sure that this allows passing `-E` multiple times.

Reply:
The type is the input type, before the action of the callback.
Setting it to "dict" would require the definition of what is a dict in OptParser, how to parse it, ... and duplicate most of the code already present. 
When not defining it, the callback gets None instead of "" and most of them fail in parsing the string.

I am not very familiar with cythonize in the command line but apparently it works now as well.